code-challenges.test.js

```diff
@@ -14,38 +14,138 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+// Create Jest test using syntax example from syllubus with describe, it, and expect statements.
+describe("occupationString", () => {
+
+  const people = [
+    { name: "ford prefect", occupation: "a hitchhiker" },
+    { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+    { name: "arthur dent", occupation: "a radio employee" }
+  ]
+  // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+    it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+
+      // set expect statement to equal an array with a sentence about each person with their name capitalized.
+      expect(occupationString(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+
+    })
+  })
+  // Note: running Jest produces desired Red message that function is not defined
+
 
 
 // b) Create the function that makes the test pass.
 
+  //create a function called occupationString
+const occupationString = (array) => {
+  // create a return array
+  let final = []
+  // create an array to push names into
+  let nameArray = []
+  // create a string to convert from the names array
+  let nameString = ""
+  // iterate over the original array
+  for (let i=0; i < array.length; i++){
+  // split the value, accessed at name, into an array, for each index
+    let upcaseNames = array[i].name.split(" ")
+  //   iterate over the array that contains the person's full name
+    for (let i=0; i < upcaseNames.length; i++){
+  //uppercase index 0, of each index, and slice the rest of the word to add to index 0
+      upcaseNames[i] = upcaseNames[i].charAt(0).toUpperCase() + upcaseNames[i].slice(1)
+  // push each uppercased name into the names array
+      nameArray.push(upcaseNames[i])
+  // rejoin the array into a string for proper display
+      nameString = nameArray.join(" ")
+    }
+  // push a string interpolation using the name string and occupation string, into the final array
+    final.push(`${nameString} is ${array[i].occupation}.`)
+  // reset the array to an empty array so it doesn't repeate info
+    nameArray = []
+  }
+ // return the final array. wow, that's a lot of pseudocoding lol
+  return final
+}
```

Comment:
Wooo look at that. Very systematic. I am very glad you took the time to explain your process.